java/java.hints/src/org/netbeans/modules/java/hints/suggestions/ExpectedTypeResolver.java

```diff
@@ -1388,4 +1391,16 @@ private static void addTypeAndReplaceMoreSpecific(CompilationInfo info, Collecti
     public List<? extends TypeMirror> visitUses(UsesTree node, Object p) {
         return null;
     }
+   
+    public List<? extends TypeMirror> visitBindingPattern(BindingPatternTree bpt, Object p) {
```

Comment:
I am a little surprised this compiles, but should be safe at runtime.

Should these new methods eventually have an implementation that would produce the expected type?